internal/processing/status/get.go

```diff
@@ -74,46 +75,139 @@ func (p *Processor) contextGet(
 		return nil, errWithCode
 	}
 
-	context := &apimodel.Context{
-		Ancestors:   []apimodel.Status{},
-		Descendants: []apimodel.Status{},
-	}
-
 	parents, err := p.state.DB.GetStatusParents(ctx, targetStatus)
 	if err != nil {
 		return nil, gtserror.NewErrorInternalError(err)
 	}
 
+	var ancestors []*apimodel.Status
 	for _, status := range parents {
 		if v, err := p.filter.StatusVisible(ctx, requestingAccount, status); err == nil && v {
 			apiStatus, err := convert(ctx, status, requestingAccount)
 			if err == nil {
-				context.Ancestors = append(context.Ancestors, *apiStatus)
+				ancestors = append(ancestors, apiStatus)
 			}
 		}
 	}
 
-	sort.Slice(context.Ancestors, func(i int, j int) bool {
-		return context.Ancestors[i].ID < context.Ancestors[j].ID
+	slices.SortFunc(ancestors, func(lhs, rhs *apimodel.Status) int {
+		return strings.Compare(lhs.ID, rhs.ID)
 	})
 
 	children, err := p.state.DB.GetStatusChildren(ctx, targetStatus.ID)
 	if err != nil {
 		return nil, gtserror.NewErrorInternalError(err)
 	}
 
+	var descendants []*apimodel.Status
 	for _, status := range children {
 		if v, err := p.filter.StatusVisible(ctx, requestingAccount, status); err == nil && v {
 			apiStatus, err := convert(ctx, status, requestingAccount)
 			if err == nil {
-				context.Descendants = append(context.Descendants, *apiStatus)
+				descendants = append(descendants, apiStatus)
 			}
 		}
 	}
 
+	TopoSort(descendants, targetStatus.AccountID)
+
+	context := &apimodel.Context{
```

Comment:
You can make this a little neater here (and avoid reslicing ancestors + descendants):

```suggestion
	context := &apimodel.Context{
		Ancestors:   make([]apimodel.Status, len(ancestors)),
		Descendants: make([]apimodel.Status, len(descendants)),
	}
	for i, ancestor := range ancestors {
		context.Ancestors[i] = *ancestor
	}
	for i, descendant := range descendants {
		context.Descendants[i] = *descendant
	}
```